### ISSUE TITLE
Security Review: Passphrase Change

### ISSUE THREAD

**Issue by daviddahl** [SpiderOak/crypton#380]
We need to do a security review of all new features. This one was added here: https://github.com/SpiderOak/crypton/pull/305/files

**Comment by daviddahl** [comment]
@chiiph have you seen this issue - do you know when you can fit this work in?

**Comment by chiiph** [comment]
Damn! I didn't get the notification, I was wondering what did happen to this. I'll take a look by the end of this week.

**Comment by daviddahl** [comment]
No worries! Thank You!

**Comment by chiiph** [comment]
I'm thinking of a way to use this to bump the pbkdf2 rounds, have you considered it?

**Comment by chiiph** [comment]
Should this:

https://github.com/SpiderOak/crypton/pull/305/files#diff-33cf792477dcd3874235ca22d11ed0efR175

be `newAccountData.accountId = req.session.accountId;`?

**Comment by daviddahl** [comment]
No, because we are getting the original account primary key from the query inside of this call: https://github.com/SpiderOak/crypton/pull/305/files#diff-33cf792477dcd3874235ca22d11ed0efR161

**Comment by daviddahl** [comment]
As far as bumping he PDKDF2 rounds, I have made it configurable in the change passphrase PR. Inside of Kloak, we are using 10K rounds instead of 1K

**Comment by chiiph** [comment]
> No, because we are getting the original account primary key from the query inside of this call: https://github.com/SpiderOak/crypton/pull/305/files#diff-33cf792477dcd3874235ca22d11ed0efR161

Hm, I don't follow. What stops me from changing another user's passphrase?

**Comment by daviddahl** [comment]
I think you found a bug! we should be calling getById:

https://github.com/SpiderOak/crypton/blob/master/server/lib/account.js#L75

Shit.
